### ISSUE TITLE
Exception from pscheduler-api "TypeError: 'NoneType' object is not iterable"

### ISSUE THREAD

**Issue by igarny** [perfsonar/pscheduler#645]
Hello Mark,
As pointed in Slack:
My system is pS testpoint staging.
I see these exceptions pop up regularly every 1h.

It could be I am able to match them to a test. I will give a second post about it later.

```
Jul 16 11:02:24 t2-psdev pscheduler-api INFO     Denied fd10:1::2 POST https://test1.rrze/pscheduler/tasks
Jul 16 11:02:25 t2-psdev pscheduler-api INFO     Denied fd10:1::2 POST https://test1.rrze/pscheduler/tasks
Jul 16 11:02:42 t2-psdev pscheduler-api ERROR    Exception: TypeError: 'NoneType' object is not iterable
Jul 16 11:02:42 t2-psdev pscheduler-api ERROR    Traceback (most recent call last):
Jul 16 11:02:42 t2-psdev pscheduler-api ERROR      File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1475, in full_dispatch_request
Jul 16 11:02:42 t2-psdev pscheduler-api ERROR        rv = self.dispatch_request()
Jul 16 11:02:42 t2-psdev pscheduler-api ERROR      File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1461, in dispatch_request
Jul 16 11:02:42 t2-psdev pscheduler-api ERROR        return self.view_functions[rule.endpoint](**req.view_args)
Jul 16 11:02:42 t2-psdev pscheduler-api ERROR      File "/usr/share/pscheduler/api-server/pschedulerapiserver/tasks.py", line 445, in tasks
Jul 16 11:02:42 t2-psdev pscheduler-api ERROR        for offer in tool_offers.get(participant, [{"name": "nothing"}])
Jul 16 11:02:42 t2-psdev pscheduler-api ERROR    TypeError: 'NoneType' object is not iterable
Jul 16 11:22:42 t2-psdev owampd[15352]: FILE=owampd.c, LINE=855, Control session terminated abnormally...
Jul 16 12:03:11 t2-psdev pscheduler-api INFO     Denied fd10:1::2 POST https://test1.rrze/pscheduler/tasks
Jul 16 12:03:11 t2-psdev pscheduler-api INFO     Denied fd10:1::2 POST https://test1.rrze/pscheduler/tasks
Jul 16 12:03:31 t2-psdev pscheduler-api ERROR    Exception: TypeError: 'NoneType' object is not iterable
Jul 16 12:03:31 t2-psdev pscheduler-api ERROR    Traceback (most recent call last):
Jul 16 12:03:31 t2-psdev pscheduler-api ERROR      File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1475, in full_dispatch_request
Jul 16 12:03:31 t2-psdev pscheduler-api ERROR        rv = self.dispatch_request()
Jul 16 12:03:31 t2-psdev pscheduler-api ERROR      File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1461, in dispatch_request
Jul 16 12:03:31 t2-psdev pscheduler-api ERROR        return self.view_functions[rule.endpoint](**req.view_args)
Jul 16 12:03:31 t2-psdev pscheduler-api ERROR      File "/usr/share/pscheduler/api-server/pschedulerapiserver/tasks.py", line 445, in tasks
Jul 16 12:03:31 t2-psdev pscheduler-api ERROR        for offer in tool_offers.get(participant, [{"name": "nothing"}])
Jul 16 12:03:31 t2-psdev pscheduler-api ERROR    TypeError: 'NoneType' object is not iterable
Jul 16 13:03:55 t2-psdev pscheduler-api INFO     Denied fd10:1::2 POST https://test1.rrze/pscheduler/tasks
Jul 16 13:03:56 t2-psdev pscheduler-api INFO     Denied fd10:1::2 POST https://test1.rrze/pscheduler/tasks
Jul 16 13:04:13 t2-psdev pscheduler-api ERROR    Exception: TypeError: 'NoneType' object is not iterable
Jul 16 13:04:13 t2-psdev pscheduler-api ERROR    Traceback (most recent call last):
Jul 16 13:04:13 t2-psdev pscheduler-api ERROR      File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1475, in full_dispatch_request
Jul 16 13:04:13 t2-psdev pscheduler-api ERROR        rv = self.dispatch_request()
Jul 16 13:04:13 t2-psdev pscheduler-api ERROR      File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1461, in dispatch_request
Jul 16 13:04:13 t2-psdev pscheduler-api ERROR        return self.view_functions[rule.endpoint](**req.view_args)
Jul 16 13:04:13 t2-psdev pscheduler-api ERROR      File "/usr/share/pscheduler/api-server/pschedulerapiserver/tasks.py", line 445, in tasks
Jul 16 13:04:13 t2-psdev pscheduler-api ERROR        for offer in tool_offers.get(participant, [{"name": "nothing"}])

```

**Comment by vvidic** [comment]
I see the same on my machine. AFAICT this is caused by some ``tool_offers[participant] = None`` in this line:

    for offer in tool_offers.get(participant, [{"name": "nothing"}])

Maybe this happens during package upgrade?

**Comment by mfeit-internet2** [comment]
Yep, that's a bug.  Happens when one of the participants doesn't return tool offers, probably because the server was overloaded and the request timed out.

As soon as I get the branches straightened out, I'll fix it.

**Comment by mfeit-internet2** [comment]
This was resolved by #748.
